cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -551,17 +554,18 @@ class fixed_point {
   explicit operator std::string() const
   {
     if (_scale < 0) {
-      auto const av   = std::abs(_value);
-      int64_t const n = std::pow(10, -_scale);
-      int64_t const f = av % n;
+      auto const av = detail::abs(_value);
+      Rep const n   = std::pow(10, -_scale);  // does this work for all values of __int128
```

Comment:
`scale` is an `int32_t`. In practice `scale` will typically be between 0 and -38. This is because an `__int128_t` can hold a precision of up to 38 decimal places and it really does not work to have a scale that is larger than the precision. A least for Spark it is disallowed and results in an error if you try (in Spark the scale is the negative of the cudf scale). 

```
scala> DecimalType(1, 5)
org.apache.spark.sql.AnalysisException: Decimal scale (5) cannot be greater than precision (1).
  at org.apache.spark.sql.types.DecimalType.<init>(DecimalType.scala:48)
  ... 49 elided
```

I don't know what pandas does. Python appears to have unlimited precision for `decimal.Decimal`.

`scale` can go very positive, but that is not covered by this code path (although Spark also disallows this by default).

Reply:
I have switched `std::pow` to `detail::exp10<Rep>` and added a few tests.